syncer/syncer.go

```diff
@@ -1328,22 +1380,23 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		switch ev := e.Event.(type) {
 		case *replication.RotateEvent:
 			err = s.handleRotateEvent(ev, ec)
-			if err != nil {
-				return terror.Annotatef(err, "current location %s", currentLocation)
+			if err = s.handleEventError(err, &startLocation, &currentLocation); err != nil {
```

Comment:
maybe we could break on `err != nil` and `s.handleEventError` outside the for-loop, it's OK not to change

Reply:
We may support skip-error-codes lately. And maybe continue the for loop after we meet some special err.